docs/reference/modules/discovery/quorums.asciidoc

```diff
@@ -0,0 +1,181 @@
+started manually then it's unlikely that too many nodes are started, but it's
+certainly possible to get into this situation if using a more automated
+orchestrator, particularly if the orchestrator is not resilient to failures
+such as network partitions.
+
+The initial quorum is only required the very first time a whole cluster starts
+up: new nodes joining an established cluster can safely obtain all the
+information they need from the elected master, and nodes that have previously
+been part of a cluster will have stored to disk all the information required
+when restarting.
+
+[float]
+==== Cluster maintenance, rolling restarts and migrations
+
+Many cluster maintenance tasks involve temporarily shutting down one or more
+nodes and then starting them back up again. By default Elasticsearch can remain
+available if one of its master-eligible nodes is taken offline, such as during a
+<<rolling-upgrades,rolling restart>>. Furthermore, if multiple nodes are stopped
+and then started again then it will automatically recover, such as during a
+<<restart-upgrade,full cluster restart>>. There is no need to take any further
+action with the APIs described here in these cases, because the set of master
+nodes is not changing permanently.
+
+[float]
+==== Auto-reconfiguration
+
+Nodes may join or leave the cluster, and Elasticsearch reacts by making
+corresponding changes to the voting configuration in order to ensure that the
+cluster is as resilient as possible. The default auto-reconfiguration behaviour
+is expected to give the best results in most situation. The current voting
+configuration is stored in the cluster state so you can inspect its current
+contents as follows:
+
+[source,js]
+--------------------------------------------------
+GET /_cluster/state?filter_path=metadata.cluster_coordination.last_committed_config
+--------------------------------------------------
+// CONSOLE
+
+NOTE: The current voting configuration is not necessarily the same as the set of
+all available master-eligible nodes in the cluster. Altering the voting
+configuration itself involves taking a vote, so it takes some time to adjust the
+configuration as nodes join or leave the cluster. Also, there are situations
+where the most resilient configuration includes unavailable nodes, or does not
+include some available nodes, and in these situations the voting configuration
+will differ from the set of available master-eligible nodes in the cluster.
+
+Larger voting configurations are usually more resilient, so Elasticsearch will
+normally prefer to add master-eligible nodes to the voting configuration once
+they have joined the cluster. Similarly, if a node in the voting configuration
+leaves the cluster and there is another master-eligible node in the cluster that
+is not in the voting configuration then it is preferable to swap these two nodes
+over, leaving the size of the voting configuration unchanged but increasing its
```

Comment:
It is unclear whether this "swapping over" is done automatically or whether some action is required by the user.